numpy/core/src/multiarray/methods.c

```diff
@@ -1229,18 +1229,67 @@ array_sort(PyArrayObject *self, PyObject *args, PyObject *kwds)
 {
     int axis=-1;
     int val;
-    NPY_SORTKIND sortkind = NPY_QUICKSORT;
+    NPY_SORTKIND sortkind = -1;
     PyObject *order = NULL;
+    PyObject *by = NULL;
     PyArray_Descr *saved = NULL;
     PyArray_Descr *newd;
-    static char *kwlist[] = {"axis", "kind", "order", NULL};
 
-    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|iO&O:sort", kwlist,
+
+    static char *kwlist[] = {"axis", "kind", "order", "by", NULL};
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|iO&OO:sort", kwlist,
```

Comment:
Let's make this keyword-only,
```suggestion
    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|iO&O$O:sort", kwlist,
```